tools/conformance/djxl_conformance.cc

```diff
@@ -106,8 +125,18 @@ bool SaveNPYArray(const char* filename, const ImageArray& arr) {
     WRITE_TO_FILE(ss.str().data(), ss.str().size());
   }
 
-  for (const auto& frame : arr.frames) {
-    WRITE_TO_FILE(frame.data(), frame.size());
+  // interleave the samples from color and extra channels
+  for (size_t f = 0; f < arr.frames.size(); ++f) {
+    size_t pos = 0;
+    for (size_t y = 0; y < arr.ysize; ++y) {
+      for (size_t x = 0; x < arr.xsize; ++x, pos += sizeof(float)) {
```

Comment:
it might greatly improve performance (and clarity) to interleave in a std::vector<float> of num_color_channels + num_extra_chanells and then write that, even maybe a whole row.

Reply:
I'd say this can stay as-is for now, since the frames and ec_frames array layout makes sense from the input side, the API outputs the data in that form. So the API writes to the interleaved color and separate extra channel form, then this loop converts it into the numpy array form, I think this is good to stay as-is.

Reply:
I mean to do the interleave in a buffer here in this for loop, instead of calling write() all the time. Before we had 1 call to write() now we have one per pixel per ec channel.